pkg/service/one2onerestore/worker_validate_test.go

```diff
@@ -0,0 +1,288 @@
+// Copyright (C) 2025 ScyllaDB
+
+package one2onerestore
+
+import (
+	"strings"
+	"testing"
+
+	gocmp "github.com/google/go-cmp/cmp"
+)
+
+func TestMapTargetHostToSource(t *testing.T) {
+	testCases := []struct {
+		name string
+
+		nodeMappings []nodeMapping
+		targetHosts  []Host
+		expected     map[string]Host
+		expectedErr  bool
+	}{
+		{
+			name: "All hosts have mappings",
+			nodeMappings: []nodeMapping{
```

Comment:
These tests are OK.
But you should verify it bit broader way.
Please keep `validateClusters` as the method you test. Most likely it will require to use integration tests instead of this ones.